contracts/libraries/OFTTransportAdapter.sol

```diff
@@ -0,0 +1,91 @@
+ * @custom:security-contact bugs@across.to
+ */
+contract OFTTransportAdapter {
+    using SafeERC20 for IERC20;
+    using AddressToBytes32 for address;
+
+    bytes public constant EMPTY_MSG_BYTES = new bytes(0);
+    address public constant ZERO_ADDRESS = address(0);
+
+    // USDT address on current chain.
+    IERC20 public immutable usdt;
+    // Mailbox address for OFT cross-chain transfers.
+    IOFT public immutable oftMessenger;
+
+    /**
+     * @notice The destination endpoint id in the OFT messaging protocol.
+     * @dev Source https://docs.layerzero.network/v2/developers/evm/technical-reference/deployed-contracts.
+     * @dev Can also be found on target chain OFTAdapter -> endpoint() -> eid().
+     */
+    uint32 public immutable dstEid;
+
+    /**
+     * @notice intiailizes the OFTTransportAdapter contract.
+     * @param _usdt USDT address on the current chain.
+     * @param _oftMessenger OFTAdapter contract to bridge to the other chain. If address is set to zero, OFT bridging will be disabled.
+     * @param _dstEid The endpoint ID that OFT will transfer funds to.
+     */
+    constructor(IERC20 _usdt, IOFT _oftMessenger, uint32 _dstEid) {
+        usdt = _usdt;
+        oftMessenger = _oftMessenger;
+        dstEid = _dstEid;
+    }
+
+    /**
+     * @notice Returns whether or not the OFT bridge is enabled.
+     * @dev If the IOFT is the zero address, OFT bridging is disabled.
+     */
+    function _isOFTEnabled(address _token) internal view returns (bool) {
+        return address(oftMessenger) != address(0) && address(usdt) == _token;
+    }
+
+    /**
+     * @notice transfers usdt to destination endpoint
+     * @param _to address to receive a trasnfer on the destination chain
+     * @param _amount amount to send
+     */
+    function _transferUsdt(address _to, uint256 _amount) internal {
+        bytes32 to = _to.toBytes32();
+
+        // todo: should probably just comment these 2 vars and use `_amount` in send call below
+        // `amountLD` and `minAmountLD` have a subtle relationship. OFT "removes dust" on .send, which should not affect USDT transfer because of how OFT works internally with 6-decimal tokens.
+        // Setting these two equal protects us from dust subtraction on the OFT side. If any dust is subtracted, the later .send should revert. Should be cautious with this logic.
+        uint256 amountLD = _amount;
+        uint256 minAmountLD = _amount;
+
+        SendParam memory sendParam = SendParam(
+            dstEid,
+            to,
+            amountLD,
+            minAmountLD,
+            EMPTY_MSG_BYTES,
+            EMPTY_MSG_BYTES,
+            EMPTY_MSG_BYTES
+        );
+
+        MessagingFee memory fee = oftMessenger.quoteSend(sendParam, false);
+
+        // todo: here, we're relying on `oftMessenger` to use the full allowance. Otherwise, next .safeApprove will revert.
+        usdt.safeApprove(address(this), _amount);
```

Comment:
Can't we check the .send() return value and make sure it uses exactly what we expect?